mailbox/plugin/quota-mailing/src/test/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerTest.java

```diff
@@ -0,0 +1,679 @@
+    default void shouldUpdateCountChangesWhenAbove(QuotaThresholdChangesStore store) {
+        FixedInstantSupplier instantSupplier = new FixedInstantSupplier();
+        QuotaMailingListener testee = new QuotaMailingListener(mailetContext(), MemoryUsersRepository.withVirtualHosting(), store, instantSupplier);
+        testee.configure(new QuotaMailingListenerConfiguration(SINGLE_THRESHOLD, GRACE_PERIOD));
+
+
+        QuotaThresholdChange oldChange = new QuotaThresholdChange(QuotaThreshold.ZERO,
+            instantSupplier.now().minus(Duration.ofDays(6)));
+        store.persistQuotaCountThresholdChange(BOB_USER, oldChange);
+
+        testee.event(new MailboxListener.QuotaUsageUpdatedEvent(new MockMailboxSession(BOB),
+            QuotaRoot.quotaRoot("any", Optional.empty()),
+            Quota.<QuotaCount>builder()
+                .used(QuotaCount.count(55))
+                .computedLimit(QuotaCount.count(100))
+                .build(),
+            Quota.<QuotaSize>builder()
+                .used(QuotaSize.size(40))
+                .computedLimit(QuotaSize.size(100))
+                .build()));
+
+        assertThat(store.retrieveQuotaCountThresholdChanges(BOB_USER))
+            .isEqualTo(new QuotaThresholdChanges(oldChange,
+                new QuotaThresholdChange(new QuotaThreshold(0.5), instantSupplier.now())));
+    }
+
+    @Test
+    default void shouldUpdateCountChangesWhenAboveButRecentlyOverpasses(QuotaThresholdChangesStore store) {
+        FixedInstantSupplier instantSupplier = new FixedInstantSupplier();
+        QuotaMailingListener testee = new QuotaMailingListener(mailetContext(), MemoryUsersRepository.withVirtualHosting(), store, instantSupplier);
+        testee.configure(new QuotaMailingListenerConfiguration(SINGLE_THRESHOLD, GRACE_PERIOD));
+
+
+        QuotaThresholdChange oldChange1 = new QuotaThresholdChange(new QuotaThreshold(0.5),
+            instantSupplier.now().minus(Duration.ofHours(12)));
+        QuotaThresholdChange oldChange2 = new QuotaThresholdChange(QuotaThreshold.ZERO,
+            instantSupplier.now().minus(Duration.ofHours(6)));
+        store.persistQuotaCountThresholdChange(BOB_USER, oldChange1);
+        store.persistQuotaCountThresholdChange(BOB_USER, oldChange2);
+
+        testee.event(new MailboxListener.QuotaUsageUpdatedEvent(new MockMailboxSession(BOB),
+            QuotaRoot.quotaRoot("any", Optional.empty()),
+            Quota.<QuotaCount>builder()
+                .used(QuotaCount.count(55))
+                .computedLimit(QuotaCount.count(100))
+                .build(),
+            Quota.<QuotaSize>builder()
+                .used(QuotaSize.size(40))
+                .computedLimit(QuotaSize.size(100))
+                .build()));
+
+        assertThat(store.retrieveQuotaCountThresholdChanges(BOB_USER))
+            .isEqualTo(new QuotaThresholdChanges(oldChange1, oldChange2,
+                new QuotaThresholdChange(new QuotaThreshold(0.5), instantSupplier.now())));
+    }
```

Comment:
maybe I missed it, but I'd like to see a test with 2 configured threshold, then one mail sent (for the first threshold), and one other mail sent when the second threshold is reached.